app/helpers/url_helper.rb

```diff
@@ -0,0 +1,11 @@
+module UrlHelper
+  def add_param_to_url(url, name, value)
+    return url if value.blank?
+
+    if url.include? "?"
```

Comment:
This is already much neater...
So won't block on my reserach... but I'm sure this is a problem something in Ruby standard lib will have solved.

Reply:
I asked around [this is the best I could find](https://freethreads.net/2019/09/09/adding-query-params-to-a-url-in-ruby-with-http/):

```
params = { 'id' => '12' }
 
path = 'http://localhost/'
uri = URI(path)
 
uri.query = URI.encode_www_form(params)
```